inloop/solutions/views.py

```diff
@@ -126,3 +138,24 @@ def get_context_data(self):
     def get_object(self, **kwargs):
         self.solution = get_object_or_404(Solution, pk=kwargs["id"])
         return self.solution
+
+
+class SolutionFileView(LoginRequiredMixin, View):
+    def get(self, request, **kwargs):
+        task = Task.objects.published().filter(slug=kwargs["slug"])
+        solution = get_object_or_404(
+            Solution, author=self.request.user, task=task, scoped_id=kwargs["scoped_id"]
+        )
+        for solutionfile in solution.solutionfile_set.all():
```

Comment:
I have to correct myself, instead of `filter()` it is even better to use `get()`, because it throws an error if more than one file matches. As this should never happen, it is a perfect *fail fast* bug detector for us :-)

The filter criteria can use the `file` field and the `__endswith` operator. Together, something like this here:

```python
last_part = "/" + title
upfile = solution.solutionfile_set.get(file__endswith=last_part)
```

The slash is prepended because otherwise the search is not precise enough (think about what happens with `endswith="Item.java"` if we have two files `BasicItem.java` and `Item.java`).